File: idk/projects/idk/network/Address.cpp
#include "stdafx.h"
#include "Address.h"

namespace idk
{
	Address::operator string() const
	{
		char buf[32];
		if (port != static_cast<unsigned short>(-1))
			sprintf_s(buf, "%d.%d.%d.%d:%d", a, b, c, d, port);
		else
			sprintf_s(buf, "%d.%d.%d.%d", a, b, c, d);
		return buf;
	}
	bool Address::operator<(const Address& rhs) const
	{
		for (int i = 0; i < 4; ++i)
		{
			if (nums[i] < rhs.nums[i])
				return true;
			if (nums[i] > rhs.nums[i])
				return false;
		}

		return port < rhs.port;
	}
}

File: idk/projects/wrapper/Electron/Address.cs
namespace idk
{
    public struct Address
    {
        public byte a, b, c, d;
        public short port;

        public Address(int _a, int _b, int _c, int _d)
        {
            a = (byte) _a;
            b = (byte) _b;
            c = (byte) _c;
            d = (byte) _d;
            port = 0;
        }
    }
}
